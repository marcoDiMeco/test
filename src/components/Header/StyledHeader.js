import styled from 'styled-components';

const StyledHeaderContainer = styled.div`
  background-color: white;
  width: 100%;
  padding-top: 30px;
  padding-bottom: 30px;
  position: fixed;
  top: 0;
  box-shadow: 0px 0px 5px #cbcbcb;
`;

const StyledHeaderContent = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  padding-left: 20px;
  padding-right: 20px;

  a {
    color: #464646;
    font-size: 16px;
    margin-left: 20px;
    margin-right: 20px;
    font-weight: 800;
    text-transform: uppercase;
  }

  a.selected {
    color: #1f88ff;
  }
`;

export {
  StyledHeaderContainer,
  StyledHeaderContent
};

