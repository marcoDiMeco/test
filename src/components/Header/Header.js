import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { AMOUNT_VALUE } from "../../constants";
import { StyledHeaderContainer, StyledHeaderContent } from "./StyledHeader";

const Header = ({ getUsers }) => {
  return (
    <StyledHeaderContainer>
      <StyledHeaderContent>
        <NavLink
          exact
          to="/list1"
          activeClassName="selected"
          onClick={() => getUsers(AMOUNT_VALUE.first)}
        >
          List 1
        </NavLink>
        <NavLink
          exact
          to="/list2"
          activeClassName="selected"
          onClick={() => getUsers(AMOUNT_VALUE.second)}
        >
          List 2
        </NavLink>
      </StyledHeaderContent>
    </StyledHeaderContainer>
  );
};

Header.propTypes = {
  getUsers: PropTypes.func,
};

export default Header;
