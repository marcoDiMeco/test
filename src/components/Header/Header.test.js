import React from "react";
import renderer from "react-test-renderer";
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";
import Header from "./";

const history = createBrowserHistory();

describe("Header component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <Header />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
