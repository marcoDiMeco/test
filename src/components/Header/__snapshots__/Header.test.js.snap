// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div
  className="sc-AxjAm juZLGC"
>
  <div
    className="sc-AxirZ gyqzDN"
  >
    <a
      aria-current={null}
      href="/list1"
      onClick={[Function]}
    >
      List 1
    </a>
    <a
      aria-current={null}
      href="/list2"
      onClick={[Function]}
    >
      List 2
    </a>
  </div>
</div>
`;
